core/src/main/java/org/jruby/util/Pack.java

```diff
@@ -935,7 +935,7 @@ private static RubyString unpackBase46Strict(ThreadContext context, ByteList inp
                 out[index++] = (byte)((b << 4 | c >> 2) & 255);
             }
         }
-        return newString(context, new ByteList(out, 0, index));
+        return newString(context, new ByteList(out, 0, index, false));
```

Comment:
Passing `false` here avoids an unnecessary array copy in the ByteList constructor. The `out` array isn't otherwise exposed outside this method, so avoiding the copy seems safe to me.